main.nf

```diff
@@ -198,72 +249,328 @@ process get_software_versions {
 
     script:
-    """
-    fastqc -q $reads
-    """
+     """
+     ${openms_path}/PeptideIndexer -in ${id_file} -out ${ID_idx}_idx.idXML -threads ${params.num_threads} -fasta ${fasta_decoy} -decoy_string DECOY -enzyme:specificity none
+     """
+
 }
 
 
+/*
+ * STEP 4 - calculate fdr for id based alignment
+ */
+process calculate_fdr_for_idalignment {
+    publishDir "${results_path}/"
+ 
+    input:
+     set ID_idx_fdr, file(id_file_idx) from id_files_idx
+
+    output:
+     set ID_idx_fdr, file("${ID_idx_fdr}_fdr.idXML") into id_files_idx_fdr
+
+    script:
+     """
+     ${openms_path}/FalseDiscoveryRate -in ${id_file_idx} -out ${ID_idx_fdr}_fdr.idXML -threads ${params.num_threads}
+     """
+
+}
+
 
 /*
- * STEP 2 - MultiQC
+ * STEP 5 - filter fdr for id based alignment
  */
-process multiqc {
-    publishDir "${params.outdir}/MultiQC", mode: 'copy'
+process filter_fdr_for_idalignment {
+    publishDir "${results_path}/"
+ 
+    input:
+     set ID_idx_fdr_filtered, file(id_file_idx_fdr) from id_files_idx_fdr
+
+    output:
+     set ID_idx_fdr_filtered, file("${ID_idx_fdr_filtered}_filtered.idXML") into id_files_idx_fdr_filtered
+
+    script:
+     """
+     ${openms_path}/IDFilter -in ${id_file_idx_fdr} -out ${ID_idx_fdr_filtered}_filtered.idXML -threads ${params.num_threads} -score:pep 0.05  -remove_decoys
+     """
+
+}
+
+
+/*
+ * STEP 6 - compute alignment rt transformation
+ */
+process align_ids {
+   publishDir "${results_path}/"
 
     input:
-    file multiqc_config
-    file ('fastqc/*') from fastqc_results.collect()
-    file ('software_versions/*') from software_versions_yaml
-    file workflow_summary from create_workflow_summary(summary)
+     file id_names from id_files_idx_fdr_filtered.collect{it[1]}
 
     output:
-    file "*multiqc_report.html" into multiqc_report
-    file "*_data"
+     file '*.trafoXML' into id_files_trafo_mzml, id_files_trafo_idxml
 
     script:
-    rtitle = custom_runName ? "--title \"$custom_runName\"" : ''
-    rfilename = custom_runName ? "--filename " + custom_runName.replaceAll('\\W','_').replaceAll('_+','_') + "_multiqc_report" : ''
-    """
-    multiqc -f $rtitle $rfilename --config $multiqc_config .
-    """
+     def out_names = id_names.collect { it.baseName+'.trafoXML' }.join(' ')
+     """
+     ${openms_path}/MapAlignerIdentification -in $id_names -trafo_out $out_names
+     """
+
 }
 
 
+/*
+ * STEP 7 - align mzML files using trafoXMLs
+ */
+process align_mzml_files {
+    publishDir "${results_path}/"
+
+    input:
+     set ID_trafo_mzml, file(id_file_trafo) from id_files_trafo_mzml
+     set mzmlID_align, file(mzml_file_align) from mzmlfiles_align
+
+    output:
+     set mzmlID_align, file("${mzmlID_align}_aligned.mzML") into mzml_files_aligned
+
+    script:
+     """
+     ${openms_path}/MapRTTransformer -in ${mzml_file_align} -trafo_in ${id_file_trafo} -out ${mzmlID_align}_aligned.mzML -threads ${params.num_threads}
```

Comment:
Simply use `task.cpus` which allows us to fine tune the parameters. Don't use separate params definitions if there is something provided by nextflow already ( params.cpus is the one you'd want to use) :-)